File: 10-databases/week5/main.py
import psycopg2

db_map = {"fname": "Name", "lname": "Last Name", "birth_date": "Bday", "sex": "sex",
          "dname": "Department Name", "dlocation": "Address"}

conn = None 
while(True):

    # This doesnt work so gpt a solution as uni so shit it cant even teach me basics
    sql = """INSERT INTO employee(fname,minit,lname,ssn,bith_date, address,sex,
            salary,super_ssn,dno) VALUES%s;"""
    insert_employee_row = ("new_fname", "C", "new_lname", 112233446,'1966-1-12',
                           "2342 May, Atlanta, GA",'M',40000.00, 111111100 ,6)

    conn = None 
    try:
        # Connects to db
        conn = psycopg2.connect("dbname=prac_3 user=lenny password=")
        cursor = conn.cursor()
        
        # Execute insert command 
        cursor.execute(sql,(insert_employee_row))
        conn.commit()

        # Gets user query
        inp: str = input("enter an sql command: ") 
        cursor.execute(inp)

        results = cursor.fetchall()

        # Gets col names from query
        colnames: list = [col[0] for col in cursor.description] 
        columns: str = ""
        for name in colnames:
            columns += db_map[name] + "\t"

        # displays query
        print(columns)
        for row in results:
            row_string: str = ""
            for var in row:
                row_string += var + "\t"
            print(row_string)
        print("Fin")

    except (Exception, psycopg2.DatabaseError) as error:
        print("Error connecting to database", error)

    finally:
        if conn:
            conn.close()
            print("Connection closed")

File: 10-databases/week5/main.java
import java.beans.Statement;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;

public class main {
    public static void main(String[] args) throws Exception{

        Connection conn = null;
        try {
            Class.forName("org.postgresql.Driver");
            String url = "jdbc:postgresql://localhost/lenny";
            conn = DriverManager.getConnection(url);
        }
        catch (ClassNotFoundException e) {
            e.printStackTrace();
            System.exit(1);
        }
        catch (SQLException e) {
            e.printStackTrace();
            System.exit(2);
        }

        System.out.println("Database things\n");
        
        String query = "SELECT fname, lname FROM employee";
        Statement stmt = null;

        try {
            stmt = conn.createStatement();
            ResultSet rs = stmt.executeQuery(query);

            while (rs.next()) {
                String fname = rs.getString("fname");
                String lname = rs.getString("lname");
                System.out.println(fname, lname);
            }
        }
        catch (SQLException e) {
            System.out.println(e);
            conn.close();
            System.exit(1);
        }

        System.out.println("Success bitch");
        conn.close();  
    }
}
